x-pack/plugins/ml/public/register_helper/register_search_links/search_deep_links.ts

```diff
@@ -8,203 +8,286 @@
-  id: 'modelManagement',
-  title: i18n.translate('xpack.ml.deepLink.modelManagement', {
-    defaultMessage: 'Model Management',
-  }),
-  path: `/${ML_PAGES.TRAINED_MODELS_MANAGE}`,
-  deepLinks: [
-    {
-      id: 'nodesOverview',
-      title: i18n.translate('xpack.ml.deepLink.trainedModels', {
-        defaultMessage: 'Trained Models',
-      }),
-      path: `/${ML_PAGES.TRAINED_MODELS_MANAGE}`,
-    },
-    {
-      id: 'nodes',
-      title: i18n.translate('xpack.ml.deepLink.nodes', {
-        defaultMessage: 'Nodes',
-      }),
-      path: `/${ML_PAGES.NODES}`,
-    },
-  ],
-};
-
-const MEMORY_USAGE_DEEP_LINK: AppDeepLink<LinkId> = {
-  id: 'memoryUsage',
-  title: i18n.translate('xpack.ml.deepLink.memoryUsage', {
-    defaultMessage: 'Memory Usage',
-  }),
-  path: `/${ML_PAGES.MEMORY_USAGE}`,
-};
-
-const DATA_VISUALIZER_DEEP_LINK: AppDeepLink<LinkId> = {
-  id: 'dataVisualizer',
-  title: i18n.translate('xpack.ml.deepLink.dataVisualizer', {
-    defaultMessage: 'Data Visualizer',
-  }),
-  path: `/${ML_PAGES.DATA_VISUALIZER}`,
-};
-
-const FILE_UPLOAD_DEEP_LINK: AppDeepLink<LinkId> = {
-  id: 'fileUpload',
-  title: i18n.translate('xpack.ml.deepLink.fileUpload', {
-    defaultMessage: 'File Upload',
-  }),
-  keywords: ['CSV', 'JSON'],
-  path: `/${ML_PAGES.DATA_VISUALIZER_FILE}`,
-};
-
-const INDEX_DATA_VISUALIZER_DEEP_LINK: AppDeepLink<LinkId> = {
-  id: 'indexDataVisualizer',
-  title: i18n.translate('xpack.ml.deepLink.indexDataVisualizer', {
-    defaultMessage: 'Index Data Visualizer',
-  }),
-  path: `/${ML_PAGES.DATA_VISUALIZER_INDEX_SELECT}`,
-};
-
-const SETTINGS_DEEP_LINK: AppDeepLink<LinkId> = {
-  id: 'settings',
-  title: i18n.translate('xpack.ml.deepLink.settings', {
-    defaultMessage: 'Settings',
-  }),
-  path: `/${ML_PAGES.SETTINGS}`,
-  deepLinks: [
-    {
-      id: 'calendarSettings',
-      title: i18n.translate('xpack.ml.deepLink.calendarSettings', {
-        defaultMessage: 'Calendars',
-      }),
-      path: `/${ML_PAGES.CALENDARS_MANAGE}`,
-    },
-    {
-      id: 'filterListsSettings',
-      title: i18n.translate('xpack.ml.deepLink.filterListsSettings', {
-        defaultMessage: 'Filter Lists',
-      }),
-      path: `/${ML_PAGES.SETTINGS}`, // Link to settings page as read only users cannot view filter lists.
-    },
-  ],
-};
-
-const NOTIFICATIONS_DEEP_LINK: AppDeepLink<LinkId> = {
-  id: 'notifications',
-  title: i18n.translate('xpack.ml.deepLink.notifications', {
-    defaultMessage: 'Notifications',
-  }),
-  path: `/${ML_PAGES.NOTIFICATIONS}`,
-};
-
-export function getDeepLinks(isFullLicense: boolean) {
+function getOverviewLinkDeepLink(mlCapabilities: MlCapabilities): AppDeepLink<LinkId> {
+  return {
+    id: 'overview',
+    title: i18n.translate('xpack.ml.deepLink.overview', {
+      defaultMessage: 'Overview',
+    }),
+    path: `/${ML_PAGES.OVERVIEW}`,
+    navLinkStatus:
+      mlCapabilities.isADEnabled && mlCapabilities.isDFAEnabled && mlCapabilities.isNLPEnabled
```

Comment:
I wonder if we should require all these capabilities for the Overview page? Perhaps we can hide some panels if certain capabilities are missing?

Reply:
The overview page is not currently available in serverless.
Having all three of these feature capabilities required is essentially like saying `isServerless === false`.
This may change if one of the projects decides they want the overview page added, in which case, yes we will need to hide/show panels based on these feature capabilities.

Reply:
Would be good to add a comment or have a util function for now for better clarity if that's the case.

Reply:
updated in https://github.com/elastic/kibana/pull/159433/commits/c29461ae8a78500890a6fb0ca638aa51a449d80b

Reply:
It is possible to access the overview page from the breadcrumbs.
Because of this I've [adjusted](https://github.com/elastic/kibana/pull/159433/commits/82a51a71339a83a1785a45536dba3b62cc41e8e0) the overview page to hide panels based on feature enablement. 
I've added note the the [meta issue](https://github.com/elastic/kibana/issues/160891) for this this task to say that further decisions will be needed by the various serverless plugins as to whether they want an ML overview page included. If they decided they do they can add a nav menu item in and we can adjust these deep links checks.

For now I'm going to keep the overview page out of the nav menu because only the Search plugin will see it, and for that plugin in the overview page is completely blank as it has AD and DFA turned off.